agent/stats/queue.go

```diff
@@ -101,7 +100,7 @@ func (queue *Queue) add(rawStat *ContainerStats) {
 			stat.CPUUsagePerc = 100 * cpuUsageSinceLastStat / timeSinceLastStat
 		}
 
-		if queue.maxSize == queueLength {
+		if queueLength >= queue.maxSize {
```

Comment:
if we always remove first element when queue is full, queueLength will never be > maxSize, so why is this change (from == to >=) necessary?

Reply:
it's just safer this way since we don't have any guarantees that nobody else will ever modify queue.maxSize